crates/apollo-mcp-server/src/runtime/graphos.rs

```diff
@@ -6,28 +6,50 @@ use apollo_mcp_registry::{
 };
 use apollo_mcp_server::errors::ServerError;
 use schemars::JsonSchema;
-use serde::Deserialize;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
 use url::Url;
 
 const APOLLO_GRAPH_REF_ENV: &str = "APOLLO_GRAPH_REF";
 const APOLLO_KEY_ENV: &str = "APOLLO_KEY";
 
+fn apollo_uplink_endpoints_deserializer<'de, D>(deserializer: D) -> Result<Vec<Url>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum UrlListOrString {
+        List(Vec<Url>),
+        String(String),
+    }
+
+    match UrlListOrString::deserialize(deserializer)? {
+        UrlListOrString::List(urls) => Ok(urls),
+        UrlListOrString::String(s) => s
+            .split(',')
+            .map(|v| Url::parse(v.trim()).map_err(D::Error::custom))
```

Comment:
Do we want to add context to the error here that it failed while trying to decode the uplink list?